### ISSUE TITLE
Fix positionig of Related Articles on Article Pages

### ISSUE THREAD

**Issue by beccaklam** [mozilla/foundation.mozilla.org#9664]
Follow up ticket from #9542:

Currently related articles look like:
<img width="758" alt="Screenshot 2022-11-09 at 11 16 37 AM" src="https://user-images.githubusercontent.com/20977376/200921555-095068e2-6626-4982-b58d-45a817b30ec8.png">

But should look like this:
![image](https://user-images.githubusercontent.com/20977376/200921501-3f4d9265-c779-487d-b1fb-c17cbc7a6cf9.png)

**Context:**
There was a conversation between Simon, Nancy and I where we decided we would have the related articles at the end of an article but _only_ in the niche situation where there was very little content (e.g. one photo no paragraph block). It looks like the niche fix has been implemented for all cases though and needs to be fixed so that it only appears in the niche edge case.

We also need more spacing between the email CTA box and the "What to read next" section:
![image](https://user-images.githubusercontent.com/20977376/200922135-3040912f-0179-483c-a5fe-3628412afea0.png)

**Comment by nancyt1** [comment]
This is also happening for some reason, especially when there's line breaks in the paragraph block.
 
<img width="1680" alt="Screen Shot 2022-11-14 at 12 44 27 PM" src="https://user-images.githubusercontent.com/74557207/201788375-3ad9d74e-fc19-47e4-bb07-74be595a9163.png">

**Comment by mmmavis** [comment]
Hello @nancyt1 Let me see if I can reproduce the issues. Could you provide link to the page?

**Comment by nancyt1** [comment]
@mmmavis You can test it on this [page](https://foundation.mozilla.org/en/privacynotincluded/articles/getting-private-mental-health-and-prayer-apps-rush-to-revise-privacy-policies/), Simon set it up so that if the first or second paragraph block is too short the Related block will move to the bottom. But it gets funky if there's line breaks or sentences aren't long enough.

**Comment by mmmavis** [comment]
@nancyt1 Thanks!

@beccaklam 

> There was a conversation between Simon, Nancy and I where we decided we would have the related articles at the end of an article but only in the niche situation where there was very little content (e.g. one photo no paragraph block). It looks like the niche fix has been implemented for all cases though and needs to be fixed so that it only appears in the niche edge case.

I'm not familiar with recent updates made to PNI. Could you elaborate what you meant by `niche situation where there was very little content (e.g. one photo no paragraph block)`? It would be super helpful if you could point me to design on Figma as well. Thanks!

**Comment by mmmavis** [comment]
@beccaklam @nancyt1 

Did some code investigation and played around with Wagtail. Unfortunately with the page being CMS managed, it is pretty troublesome to position the "Related" section based on content length and make sure the layout show up nicely. Editors can build the page content however they want it using streamfields and options on Wagtail. It will involve lots of effort to write script that captures all possible cases and shuffle DOM elements around to achieve the desired layout. Could we consider another layout option? Something that not so content-dependent would be lovely. Ideally, we would want to avoid using JavaScript to change the ordering of the DOM elements as it will change how the content is interpreted by screen reader, search engine etc.

**Comment by beccaklam** [comment]
@mmmavis Thank you for investigating! I'm okay with going the route of a more traditional layout. @nancyt1 Would you be okay if we shifted to having the related articles like how we do on Product Review pages (i.e. on the side)?

**Comment by nancyt1** [comment]
I'm ok with that! Is there a way to have it show up towards the middle of page? Whatever the length of content may be?

**Comment by mmmavis** [comment]
Did some quick tests and it looks if we wanna to put the "related" section on the side, it will only look good for screens with width larger than 1800px. This is because video embeds(orange block in the screencap) are taking full width of the main content container. For screen width under 1800px, the side "related" section overlaps with the article contant.

This is how it looks like on 1400px wide screen. Notice the orange block (video) and pink block ("related" section") overlap with each other.
![1400pxW.png](https://images.zenhubusercontent.com/568fffb1638e718c995fe96d/de8b41c5-3f9b-4fd1-b1f1-c1c04c3b0f58)

**Comment by beccaklam** [comment]
Thank you for flagging @mmmavis ! In that case would we be able to implement an earlier breakpoint for when the related articles shifts and goes inside the main stream field? This is the [tablet](https://www.figma.com/file/GwdoUiY4fzVyeLfXsAJGPq/*Privacy-Not-Included-Website?node-id=3753%3A52397&t=xA6Lz02hP9u6dvsO-1) breakpoint and this is the [mobile](https://www.figma.com/file/GwdoUiY4fzVyeLfXsAJGPq/*Privacy-Not-Included-Website?node-id=3083%3A30676&t=xA6Lz02hP9u6dvsO-1).

**Comment by mmmavis** [comment]
Hi @beccaklam , 

For screens less than 1800px wide, do we know where in the main content we want the "related article" section to shift into? If it's something like "always after the first or n-th streamfield" then technically it should be straightforward to implement. One note is that implementing this design will change how screen readers and search engine reads the page as well.  E.g,. if we want the "related article" section to show up after the first "body" streamfield, main article content in the page will be read in the following order 
   - 1st "body" streamfield content
   - "related articles" section
   - 2nd "body" streamfield content
   - 3rd ...
   - 4th ...
   - ...

**Comment by mmmavis** [comment]
@beccaklam @nancyt1 thoughts?

**Comment by mmmavis** [comment]
Not an issue but just a concern. Having a different section inserted into the main article content could be disrupting. It could give the insert ad feel to some people. Let's try inserting it towards the middle of the content like you @beccaklam @nancyt1 suggested, and revise later if needed. Regarding screen reader - I'll apply some special HTML attributes to the "related section" to signal screen readers that it's not part of the article body content.

**Comment by mmmavis** [comment]
TL;DR
Due to design concern that can't be resolved with Tailwind limitations. We are thinking about going a different route -- creating a new streamfield block "ParagraphWithRelatedLinks" instead. This will require further discussion. Full details below.

---

On Friday I showed @nancyt1 with my implementation based on the design discussed above.

For screens wider than 1800px, because I anchored the side section to the right edge of the screen, there was design concern about the side section being too far away from the main content on extra wide screens. See the two screenshots below. First one was taken on 1800px wide screen and second one was taken on 2400px. 

![Screen Shot 2022-12-11 at 23 22 07](https://user-images.githubusercontent.com/2896608/206984663-815a6f8c-cbec-48d1-ab7e-0bfa12074382.png)

![Screen Shot 2022-12-11 at 23 22 01](https://user-images.githubusercontent.com/2896608/206984670-ccc424f3-6902-4d47-b36f-28c327b04c20.png)


There was another approach I could go to make the side section to always stick to the side of the main content (i.e., zero or suitable gap in between), however, with limitations with Tailwind I was not able to achieve this as I hoped. (e.g., Tailwind doesn't allow me to do `calc(50% + half_of_the_tw_container_width)` which I was hoping to use that value to set the side section's position).

We then started [a discussion thread on Slack](https://mozilla.slack.com/archives/C04DYP6T6VD/p1670610457144559) to see if we should go another route -- adding a new streamfield block "ParagraphWithRelatedLinks" so editors can decide if and where exactly they want the "related articles links" to show up in their articles (vs us forcing the section to show up in the middle of the article, on all articles, like what we originally planned for this ticket). Further discussion is needed before we move forward.

**Comment by nancyt1** [comment]
Hey @mmmavis  , Becca and I talked briefly about it and I think going the new streamfield block route is the best solution. How should we move forward from here? Are there other details we should discuss?

**Comment by mmmavis** [comment]
Hi @nancyt1 , thanks for updating Becca with the convo we had. As for next step, could you or Becca create a new ticket to specify what you would like the new streamfield block to be and attach design comps? Please tag Tibor and me on the ticket as well. Thank you!

**Comment by beccaklam** [comment]
@mmmavis is it okay if I close this ticket now since the work is mainly covered in #9921 and #9874?
